orttraining/orttraining/python/training/optim/config.py

```diff
@@ -0,0 +1,196 @@
+class _OptimizerConfig(object):
```

Comment:
I want to double-check that there's no scenario in which a user is expected to instantiate this base class, correct? If there is, then we should reconsider naming this class without the leading `_`. If not, we are fine.

Reply:
I don't think the user can extend this class, as each `_OptimizerConfig` class is backed by a C++ counterpart. That is one big difference between LRScheduler and _OptimizerConfig. The former is a python-only class implemented in the frontend while the latter is a C++ class exposed to python as a string. The frontend just implement a wrapper to encapsulate it better